func.go

```diff
@@ -317,10 +346,54 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 	fn.Set(v)
 }
 
+func isAllSameFloat(ty reflect.Type) bool {
+	first := ty.Field(0).Type.Kind()
```

Comment:
Doesn't this cause panic for an empty struct?

Reply:
It would but the size is always checked before entering this function so that doesn't happen. There is a test for empty return structs

Reply:
Took me a moment to realize that as well, wouldn't it be helpful to add a comment here, explaining why that's so?